code-challenges.test.js

```diff
@@ -13,40 +13,207 @@
+// Input: an array of five card values
+// Output: a boolean response
+// Problem doesn't state whether royal cards are string/characters or number values; initially treat them as values with option to refactor (J = 11, Q = 12, K = 13, A = 14)
+// Initially ignore the Joker card with the option to refactor
+// Use .reduce with a sparse array in the accumulator to count the occurences of each number in the hand (ie a hash map)
+// Use a .filter to remove non-number values
+// When max value in the array is 3 and min non-nil value in the array is 2, then there is a full hand
+
+const fullHouseCheck = (hand) => {
+      counts = hand.reduce((acc, card) => {
+            acc[card] = acc[card] + 1 || 1
+            return acc
+      }, Array(15)).filter(count => count)
+      return Math.min(...counts) === 2 && Math.max(...counts) === 3
+}
+
+// Test Suites: 1 passed, 1 total
+// Tests:       3 passed, 3 total
+// Snapshots:   0 total
+// Time:        0.579 s, estimated 1 s
+// Ran all test suites.
+// ✨  Done in 2.08s.
+
+// Alternate method using a Hash Map to track the counts of each value from the hand
+// Create a hashmap to track counts of each value
+// key = card number
+// value = count of each card with that value
+// Iterate accross the hand, updating the hash map
+// Cast the object into an array of values, flatten the values, and find the min and max card counts
+// Return true when there are two of one card and three of another
+// Otherwise return false
+
+const fullHash = (hand) => {
+      let acc = {}
+      hand.forEach(card => acc[card] = acc[card] ? acc[card] + 1 : 1)
+
+      return 3 === Math.max(...Object.values(acc))  && 2 === Math.min(...Object.values(acc)) ? true : false
+}
+
+// Test Suites: 1 passed, 1 total
+// Tests:       3 passed, 3 total
+// Snapshots:   0 total
+// Time:        0.575 s, estimated 1 s
+// Ran all test suites.
+// ✨  Done in 2.38s.
+
+// Single test shows that hashmaps with .reduce's accumulator are potentially more efficient than implementation with an object. Let's try a sparse array.
+
+const arrHash = (hand) => {
+      let acc = []
+      hand.forEach(card => acc[card] = acc[card] ? acc[card] + 1 : 1)
+      acc = acc.filter(count => count)
+      return 3 === Math.max(...acc)  && 2 === Math.min(...acc) ? true : false
+}
+
+// Test Suites: 1 passed, 1 total
+// Tests:       3 passed, 3 total
+// Snapshots:   0 total
+// Time:        0.434 s, estimated 1 s
+// Ran all test suites.
+// ✨  Done in 1.96s.
+
+// Single test shoes some improvement over object-based implementation and marginal improvement over .reduce implementation.
+
+// Test Suites: 1 passed, 1 total
+// Tests:       3 passed, 3 total
+// Snapshots:   0 total
+// Time:        0.453 s, estimated 1 s
+// Ran all test suites.
+// ✨  Done in 1.63s.
+
+// Test Suites: 1 passed, 1 total
+// Tests:       3 passed, 3 total
+// Snapshots:   0 total
+// Time:        0.304 s, estimated 1 s
+// Ran all test suites.
+// ✨  Done in 0.93s.
+
```

Comment:
This is outstanding! The consideration for better performance really shows and I also suspect something is happening that's making these tests run slightly faster each repetition.  Would be cool to look into 👨🏽‍💻